### DISCUSSION TITLE
Is it possible to hook frontend routing (rewrite path without updating url/page) in dev mode?

### DISCUSSION THREAD

**Discussion by joesonw** [ElMassimo/iles#251] [Q&A]
Say I have a incoming request `/abc.com`, but I want it to render using page `pages/store/[name].vue` underneath.
Is it possible to implement such features currently?

I tried to use `router.beforeResolve` within `enhanceApp`, but it will change the url of the page. Can it be done without changing the url?

**Comment by ouuan** [comment]
See https://iles.pages.dev/guide/routing#override-routes. It supports dynamic routes.

**Reply by ouuan** [reply]
However, in this way only `/abc.com` but not `/store/abc.com` can access this page.

**Reply by joesonw** [reply]
I currently already had dynamic routing set up. I was looking for a way to `rewrite` urls.

I think I missed another place where this is capable, in `enhanceApp`.

**Reply by ouuan** [reply]
Do you notice the hash in the URL? I mean the "Override Routes" section and the `path` field in the frontmatter.

**Reply by joesonw** [reply]
> Do you notice the hash in the URL? I mean the "Override Routes" section and the `path` field in the frontmatter.

Yeah, I wanted a dynamic rewrite. And I think `extendFrontmatter` mentioned in the tips below only works for build-time rewriting.
